erpnext/selling/doctype/sales_order/sales_order.py

```diff
@@ -64,6 +66,22 @@ def validate(self):
 		if not self.billing_status: self.billing_status = 'Not Billed'
 		if not self.delivery_status: self.delivery_status = 'Not Delivered'
 
+	def on_update_after_submit(self):
+		self.check_overdue_status()
+
+	def check_overdue_status(self):
+		overdue_conditions = [
+			self.docstatus == 1,
+			self.status not in ["On Hold", "Closed", "Completed"],
+			self.skip_delivery_note == 0,
+			flt(self.per_delivered, 6) < 100,
+			getdate(self.delivery_date) < getdate(today()),
+		]
+
+		is_overdue = all(overdue_conditions)
+		if is_overdue != self.is_overdue:
+			self.db_set("is_overdue", is_overdue)
```

Comment:
Check if `db_set` is to be used here, it triggers before_change and on_change events, also it does not commit.

Reply:
@MyuddinKhatri did you test this?

Reply:
Yes i did test this it is working fine